src/main/java/htsjdk/variant/variantcontext/writer/BCF2Writer.java

```diff
@@ -122,6 +122,10 @@
     private VCFHeader lastVCFHeaderOfUnparsedGenotypes = null;
     private boolean canPassOnUnparsedGenotypeDataForLastVCFHeader = false;
 
+    // is the header or body written to the output stream?
+    private boolean isWrittenToOutput;
```

Comment:
I think the name `outputHasBeenWritten` would be more clear.   Can you change this in all places where this variable has been added?